x-pack/plugins/security_solution/public/detection_engine/rule_management_ui/components/rules_table/upgrade_prebuilt_rules_table/upgrade_prebuilt_rules_tour.tsx

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type {
+  EuiStatelessTourStep,
+  EuiTourActions,
+  EuiTourState,
+  EuiTourStepProps,
+} from '@elastic/eui';
+import {
+  EuiButtonIcon,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiSpacer,
+  EuiText,
+  EuiTourStep,
+  useEuiTour,
+} from '@elastic/eui';
+import { noop } from 'lodash';
+import type { FC } from 'react';
+import React, { useEffect, useMemo } from 'react';
+import { NEW_FEATURES_TOUR_STORAGE_KEYS } from '../../../../../../common/constants';
+import { useKibana } from '../../../../../common/lib/kibana';
+import { useIsElementMounted } from '../rules_table/guided_onboarding/use_is_element_mounted';
+import * as i18n from './translations';
+
+export interface RulesFeatureTourContextType {
+  steps: EuiTourStepProps[];
+  actions: EuiTourActions;
+}
+
+export const PER_FIELD_UPGRADE_TOUR_ANCHOR = 'perFieldUpgradeTour';
+
+const TOUR_STORAGE_KEY = NEW_FEATURES_TOUR_STORAGE_KEYS.RULE_MANAGEMENT_PAGE;
+const TOUR_POPOVER_WIDTH = 400;
+
+const tourConfig: EuiTourState = {
+  currentTourStep: 1,
+  isTourActive: true,
+  tourPopoverWidth: TOUR_POPOVER_WIDTH,
+  tourSubtitle: '',
+};
+
+const stepsConfig: EuiStatelessTourStep[] = [
+  {
+    step: 1,
+    title: i18n.UPDATE_TOUR_TITLE,
+    content: <EuiText>{i18n.UPDATE_TOUR_DESCRIPTION}</EuiText>,
+    stepsTotal: 1,
+    children: <></>,
+    onFinish: noop,
+    maxWidth: TOUR_POPOVER_WIDTH,
+  },
+];
+
+export const PrebuiltRulesUpgradeTour: FC = () => {
+  const { storage } = useKibana().services;
+
+  const restoredState = useMemo<EuiTourState>(
+    () => ({
+      ...tourConfig,
+      ...storage.get(TOUR_STORAGE_KEY),
+    }),
+    [storage]
+  );
+
+  const [tourSteps, tourActions, tourState] = useEuiTour(stepsConfig, restoredState);
+
+  useEffect(() => {
+    const { isTourActive, currentTourStep } = tourState;
+    storage.set(TOUR_STORAGE_KEY, { isTourActive, currentTourStep });
+  }, [tourState, storage]);
+
+  const shouldShowRuleUpgradeTour = useIsElementMounted(PER_FIELD_UPGRADE_TOUR_ANCHOR);
```

Comment:
When you open any rules without closing the tour first, this happens:
![image](https://github.com/elastic/kibana/assets/5354282/c00304f4-8ffb-42f2-b2b0-18391bdd2968)
The tour gets overlayed over the flyout.

We can solve this by changing how `shouldShowRuleUpgradeTour` is calculated:
```ts
  const isTourAnchorMounted = useIsElementMounted(PER_FIELD_UPGRADE_TOUR_ANCHOR);
  const isFlyoutOpen = useIsElementMounted('updatePrebuiltRulePreview');

  const shouldShowRuleUpgradeTour = isTourAnchorMounted && !isFlyoutOpen;
```
Notice that `updatePrebuiltRulePreview` is an `id` that needs to be added to the <RuleDetailsFlyout /> component for this to work. It currently only has a `data-test-subj` with that same string, but our `useIsElementMounted` hook only works with `id`s.

Reply:
Good catch, fixed in https://github.com/elastic/kibana/pull/176767/commits/be35f698d69dd1262869ec7197dfb466800cd32b